filesystem/android_super.ksy

```diff
@@ -0,0 +1,200 @@
+meta:
+  id: android_super
+  title: Android Dynamic Partitions metadata
+  application: Android
+  file-extension: img
+  license: CC0-1.0
+  ks-version: 0.9
+  bit-endian: le
+  endian: le
+
+doc: |
+  The metadata stored by Android at the beginning of a "super" partition, which
+  is what it calls a disk partition that holds one or more Dynamic Partitions.
+  Dynamic Partitions do more or less the same thing that LVM does on Linux,
+  allowing Android to map ranges of non-contiguous extents to a single logical
+  device. This metadata holds that mapping.
+
+doc-ref:
+  - https://source.android.com/devices/tech/ota/dynamic_partitions/implement
+  - https://android.googlesource.com/platform/system/core/+/master/fs_mgr/liblp/include/liblp/metadata_format.h
+
+instances:
+  root:
+    pos: 0x1000
+    type: root
+
+types:
+  root:
+    seq:
+      - id: primary_geometry
+        size: 0x1000
+        type: geometry
+
+      - id: backup_geometry
+        type: geometry
+        size: 0x1000
+
+      - id: primary_metadata
+        size: primary_geometry.metadata_max_size
+        type: metadata
+        repeat: expr
+        repeat-expr: primary_geometry.metadata_slot_count
+
+      - id: backup_metadata
+        size: primary_geometry.metadata_max_size
+        type: metadata
+        repeat: expr
+        repeat-expr: primary_geometry.metadata_slot_count
+
+  geometry:
+    seq:
+      - id: magic
+        contents: 'gDla'
+      - id: struct_size
+        type: u4
```

Comment:
I'm using the 0x1000 length substream because whatever comes after a geometry struct is always aligned like that. I suppose I could use instances here, but an extra ~4096 bytes twice during the entire parsing doesn't seem worth the additional complexity to me.

I'd prefer not to do too much extra work on this format, since it works well for my purposes as is. Are you willing to merge this in its current state, or do you need this refactoring done first?

Reply:
> an extra ~4096 bytes twice during the entire parsing doesn't seem worth the additional complexity to me.

Yeah, you're actually right, the complexity isn't worth. And my argument about reading the 4096 byte array is actually the problem of KS (i.e. absence of https://github.com/kaitai-io/kaitai_struct/issues/44 implementation), not the spec itself. The format is OK in the current state.

I always start with a simple idea and then elaborate it in a more detail, and until the elaboration it's not clear if the idea is good or not 🙂